source/common/tcp_proxy/upstream.cc

```diff
@@ -199,6 +200,12 @@ void TcpConnPool::onPoolFailure(ConnectionPool::PoolFailureReason reason,
 
 void TcpConnPool::onPoolReady(Tcp::ConnectionPool::ConnectionDataPtr&& conn_data,
                               Upstream::HostDescriptionConstSharedPtr host) {
+  if (downstream_info_.downstreamAddressProvider().connectionID()) {
+    ENVOY_LOG(debug, "Attached upstream connection [C{}] to downstream connection [C{}]",
+              conn_data->connection().id(),
+              downstream_info_.downstreamAddressProvider().connectionID().value());
+  }
```

Comment:
why cannot do this in the `onGenericPoolReady` of `TcpProxy::Filter`?

Reply:
In ``TcpConnPool`` case it works, but for ``HttpConnPool`` the behavior is slightly different - first, ``onPoolReady`` is called to indicate network connection is ready, then it sends the ``CONNECT`` headers, then if there's a headers response ``onGenericPoolReady`` will be called. Since the attach of upstream connection to downstream connection does not depend on L7 data, I logged it at ``onPoolReady`` as it's an earlier stage.